### PR TITLE
Dev extensions

### PR DESCRIPTION
**Pull request by research11111**
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: avogadro/qtgui/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,15 @@
+find_package(Eigen3 REQUIRED)
+# Add as "system headers" to avoid warnings generated by them with
+# compilers that support that notion.
+include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
+
+cmake_minimum_required(VERSION 3.5)
```
Comment (cryos) [inline avogadro/qtgui/CMakeLists.txt:6]: None of this makes sense and is not the direction we want to move the CMake code in this project.

**Comment by research11111** [comment]
This pull request addresses Issue #1391 and includes the following enhancements:

- Sets the Python path similarly to how the `quantumInput` extension does.
- Partial support for `virtualenv`; manual path input is required from the user.
- Enables the addition of custom plugin URLs (GitHub-hosted only).
- Allows local download of the `plugins.json` file and the ability to add entries to it.
- Supports installations both with and without Conda.
- Features auto-detection of existing installations.
- Automates the installation of dependencies, either from `requirements.txt` or `pyproject.toml` (https://github.com/research11111/avogadro-rdkit/ may be used to validate).

File: avogadro/qtgui/CMakeLists.txt (hunk 1 above)
Comment (research11111) [reply avogadro/qtgui/CMakeLists.txt:6]: mistake it remove

**Comment by research11111** [comment]
add CMakeFile

**Pull request closed without merging**